### PR TITLE
Kerchunk NOS S3 files with Argo Events

### PR DESCRIPTION
**Pull request by mpiannucci**
This adds an automated kerchunking workflow running in the nextgen-dmac k8s cluster with `argo` and `argo-events`. 

In AWS, an SQS queue (`nextgen-nos-newofsobject`) is subscribed to `arn:aws:sns:us-east-1:123901341784:NewOFSObject` which is documented here https://registry.opendata.aws/noaa-ofs/ . This notification is published for every OFS Object added to the bucket. This means we get a notification in the SQS channel every time that a model run output is published. 

This PR adds `nos-sqs-source.yaml` which subscribes to the SQS channel, `nos-sqs-sensor.yaml` which triggers our kerchunking python script. 

The python script runs in a custom docker image and kerchunks the OFS netcdf file and then writes it to the `nextgen-dmac` S3 bucket. 

For now, only `dbofs` model output is acted upon to prove out the concept.

### CODE REVIEW

File: kerchunk/nos-sqs-sensor.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                  - name: message
+                    # this is the value that should be overridden
+                    value: hello world
+                templates:
+                - name: extract-sqs-message
+                  inputs:
+                    parameters:
+                    - name: message
+                  script:
+                    image: miannuccirps/kerchunk-nos:latest
```
Comment (jonmjoyce) [inline kerchunk/nos-sqs-sensor.yaml:36]: Could you push this image to a new public ECR in AWS? I've been using the annotation nextgen-dmac/<image name>

File: kerchunk/nos-sqs-sensor.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                      valueFrom:
+                        secretKeyRef:
+                          name: aws-secret
+                          key: accesskey
+                    - name: AWS_SECRET_ACCESS_KEY
+                      valueFrom:
+                        secretKeyRef:
+                          name: aws-secret
+                          key: secretkey
+                    command: [python]
```
Comment (jonmjoyce) [inline kerchunk/nos-sqs-sensor.yaml:48]: I hope we can move this Python code out of yaml!

File: kerchunk/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+Build and push the docker image:
+
+https://hub.docker.com/r/miannuccirps/kerchunk-nos
+
+```bash
+docker build -t miannuccirps/kerchunk-nos:latest .
+docker push miannuccirps/kerchunk-nos:latest
+```
```
Comment (jonmjoyce) [inline kerchunk/README.md:25]: Love the start to documenting this feature. We will want to expand this as we progress and make it easy to understand for someone like Micah.

File: kerchunk/nos-sqs-sensor.yaml (hunk 2 above)
Comment (mpiannucci) [reply kerchunk/nos-sqs-sensor.yaml:48]: I can try to pass the parameter in as a string but out of the box this allows the template to use it as a native python dictionary.

File: kerchunk/nos-sqs-sensor.yaml (hunk 1 above)
Comment (mpiannucci) [reply kerchunk/nos-sqs-sensor.yaml:36]: Yeah I'll play with that next

**Pull request merged**